### PR TITLE
Make Docker the Default Install Method #Hacktoberfest

### PR DESCRIPTION
**Pull request by cc-a**
Made Docker the preferred installation method over anaconda in the project README.

### CODE REVIEW

**Comment by mloubout** [comment]
why?

**Comment by georgebisbas** [comment]
> why?

Agreed with @ggorman ?

**Comment by FabioLuporini** [comment]
for completeness: I basically won't be able to provide any kind of support should there be any issues related to docker install 

we have pretty much 0 installation issues, I'm not sure I understand all this stress about using Docker over miniconda. What do I gain from this? What are the advantages?

super dumb question: if I'm using docker, and want to try a change to the devito code (e.g., to the notebooks), is it straightforward just as if I were using conda ?

**Comment by navjotk** [comment]
> if I'm using docker, and want to try a change to the devito code (e.g., to the notebooks), is it straightforward just as if I were using conda ?

I was under the impression that we've been through this entire discussion before and agreed to do this. As was @ggorman .

**Comment by FabioLuporini** [comment]
OK no problem

**Comment by ggorman** [comment]
As @navjotk said - we had this whole discussion before and the conclusion was docker should be the first option. Some of the rational:
- it ‘just works’ for those that just want to take devito for a quick test drive. From the OSX broken compiler problem to rouge env variables, building devito doesn’t always go as smoothly as we’d like, as clearly illustrated on Thursday. These user experience install issues are not appearing as GitHub issues but we have seen a lot on slack - and that’s from those that even go to slack (we have plenty of indirect evidence of this) so using number of GitHub issues as the sole indicator is a classic example of selection bias. 
- the docker route *must* always be fully supported because we use it for pipelines/CI and tutorials/training (binderhub). 
- not everyone is a fan of conda. There is a large community of Enthought users in geo. Our first move should not to tell them they have to change python distribution just to install our code.

**Comment by mloubout** [comment]
>the docker route must always be fully supported because we use it for pipelines/CI

These don't use the same docker build though. Pipelines has its own that is not the one from the install instructions so the recommended install is not actually tested while conda and docker (*not docker-compose* which is another all story) are indeed tested. So I would only move towards this PR if it is actually tested and explicitly explained in the readme that this is *docker-compose* not and actual available docker image of devito.

**Comment by FabioLuporini** [comment]
We agreed that this PR is on hold until someone takes responsibility of the issue that @mloubout pointed out above. 

@ggorman how do you want to proceed?

**Comment by cc-a** [comment]
Apologies, was away for a few days. Happy to take this further.

Regarding the docker vs docker-compose I'd make the case for providing instructions for just docker as:
- The volumes mounted in docker-compose.yml are convenient for development work but not needed or necessarily desirable for the end user. 
- The use of `docker-compose up` requires users to edit the yml to use a different port for the notebook server. 
- Removes a dependency

The equivalent docker invocations would be something like:
```
docker build -f docker/Dockerfile -t devito .
docker run -p 8888:8888 --rm -it devito
docker run -p 8888:8888 --rm -it devito /tests
docker run -p 8888:8888 --rm -it devito /bin/bash
```

If you're happy to move forward on that I'll look at adding some testing too.

**Comment by georgebisbas** [comment]
Closing temporarily, related to issue #817

**Comment by navjotk** [comment]
This can be reopened now that the Docker installation method is tested.

File: README.md [hunk 1]
```diff
@@ -53,6 +38,26 @@ docker-compose up devito
...
+image is built. Subsequent calls to docker-compose will be much
+quicker.
+
+Alternatively you can use the Conda package manager for installation of
+the necessary software dependencies. Please
+install either [Anaconda](https://www.continuum.io/downloads) or
+[Miniconda](https://conda.io/miniconda.html) using the instructions
+provided at the download links. You will need the Python 3 version.
+
+To install Devito, including examples, tests and tutorial notebooks,
```
Comment (FabioLuporini) [inline README.md]: this may be dropped

**Comment by navjotk** [comment]
Closing as superceded by #1116

**Pull request closed without merging**